### ISSUE TITLE
When I drag it, the error message shows "width is null"

### ISSUE THREAD

**Issue by Frank0945** [hanshengchiu/reorderables#5]
I just used ReorderableColumn to match ListView.builder.

**Comment by davidBarbieri** [comment]
Got the same error by copy/pasting the ReorderableSliverList example, i guess something has changed in some flutter update

**Comment by hanshengchiu** [comment]
> ReorderableSliverList

@davidBarbieri you meant this?
``` 
          delegate: ReorderableSliverChildBuilderDelegate(
            (BuildContext context, int index) => _rows[index],
            childCount: _rows.length
          ),
```

**Comment by hanshengchiu** [comment]
> I just used ReorderableColumn to match ListView.builder.

@Frank0945 how did you "match" them? ListView.builder creates a ListView, not a list of elements, whereas ReorderableColumn is also a list-like view and it takes a list of elements as its children.

**Comment by davidBarbieri** [comment]
> 
> 
> > ReorderableSliverList
> 
> @davidBarbieri you meant this?
> 
> ```
>           delegate: ReorderableSliverChildBuilderDelegate(
>             (BuildContext context, int index) => _rows[index],
>             childCount: _rows.length
>           ),
> ```
Hey @hanshengchiu !
Yes, i just copied al the example in a new project and got the same width error.

**Comment by hanshengchiu** [comment]
> > > ReorderableSliverList
> > 
> > 
> > @davidBarbieri you meant this?
> > ```
> >           delegate: ReorderableSliverChildBuilderDelegate(
> >             (BuildContext context, int index) => _rows[index],
> >             childCount: _rows.length
> >           ),
> > ```
> 
> Hey @hanshengchiu !
> Yes, i just copied al the example in a new project and got the same width error.
 
You may have forgotten to provide a ScrollController to the ScrollView that contains ReorderableSliverList. See https://github.com/hanshengchiu/reorderables/blob/master/example/lib/sliver_example.dart#L31

The example still works. You can uncomment sliver_example.dart#L52 and try it.

**Comment by davidBarbieri** [comment]
Looks like you're right, the exapmle works, i used a scroll controller in my app, and tried many combinations... 
Maybe the troble is caused by being inside a ScopedModelDescendant builder functions.
I'll have some more feedback in the next days

Thanks a lot for the plugin, it's just what i needed!!!!

**Comment by hanshengchiu** [comment]
closing it for now. please reopen it if this is issue.

**Comment by RyMo95** [comment]
@davidBarbieri Hello David, did you find what was causing this in your project?  I'm facing the same issue where the example works but when adding to a component in my project I get thrown this error.

Thanks!

**Comment by hanshengchiu** [comment]
@RyMo95 can you post your code?
This issue is largely related to scroll controller. I just updated the wrap example with SingleChildScrollView. FYR.

**Comment by davidBarbieri** [comment]
Hey, i will try to post the code and have a detailed error log this evening.
I commented the block and used a normal SliverList in the meanwhile, was struggling with another part of the app but i'll have a look tonight

**Comment by hanshengchiu** [comment]
if any of you is using ReorderableWrap, check out v0.2.5.

**Comment by davidBarbieri** [comment]
I'm back with my example stripped to the bare minumum, and still getting the same width = null error.
At first i thought it was the Dismissable widget wrapping all my list items, but now i'm down to real basic widgets:
This is the code that is still not working, above this there is a MaterialApp rendering the widgets from the "onGenerateRoute" prop. 
I attached a listener to the scroll controller to make sure it is working, and it is:

`
import 'package:flutter/material.dart';
import 'package:reorderables/reorderables.dart';
import '../widgets/sliver_search_input.dart';

class WorkoutsEditorScreen extends StatefulWidget {
  @override
  _WorkoutsEditorScreenState createState() => _WorkoutsEditorScreenState();
}

class _WorkoutsEditorScreenState extends State<WorkoutsEditorScreen> {
  bool isReordering = false;
  String filter = "";

  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    ScrollController _scrollController =
        PrimaryScrollController.of(context) ?? ScrollController()..addListener(() => print("listener"));
    return Scaffold(
      //floatingActionButton: _buildFABs(context),
      resizeToAvoidBottomPadding: false,
      body: SafeArea(
        child: CustomScrollView(
          controller: _scrollController,
          slivers: <Widget>[
            SliverAppBar(
              forceElevated: false,
              centerTitle: true,
              title: Text(
                "** Manage workouts **",
              ),
              floating: true,
              expandedHeight: 140,
              elevation: 4,
              pinned: true,
              flexibleSpace: FlexibleSpaceBar(
                background: Align(
                  alignment: Alignment.bottomCenter,
                  child: SliverSearchInput(
                    placeholder: "Search hint",
                    onChanged: (String value) {
                      setState(
                        () {
                          filter = value;
                        },
                      );
                    },
                  ),
                ),
                //collapseMode: CollapseMode.parallax,
              ),
            ),
            _buildWorkoutsList(),
          ],
        ),
      ),
    );
  }

  Widget _buildWorkoutsList() {
    List<Widget> _rows = List<Widget>.generate(50,
        (int index) => Text('This is sliver child $index', textScaleFactor: 2));
    return ReorderableSliverList(
      delegate: ReorderableSliverChildListDelegate(_rows),
      onReorder: (int from, int to) {},
    );
  }
}
`

This is the error log:

`I/flutter (15265): ══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
I/flutter (15265): The following assertion was thrown while handling a gesture:
I/flutter (15265): An attached scroll controller is needed. You probably forgot to attach one to the parent scroll view
I/flutter (15265): that contains this reorderable list.
I/flutter (15265): 'package:reorderables/src/widgets/reorderable_sliver.dart': Failed assertion: line 541 pos 12:
I/flutter (15265): '_scrollController.positions.isNotEmpty'
I/flutter (15265): 
I/flutter (15265): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (15265): more information in this error message to help you determine and fix the underlying cause.
I/flutter (15265): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (15265):   https://github.com/flutter/flutter/issues/new?template=BUG.md
I/flutter (15265): 
I/flutter (15265): When the exception was thrown, this was the stack:
I/flutter (15265): #2      _ReorderableSliverListState._scrollTo (package:reorderables/src/widgets/reorderable_sliver.dart:541:12)
I/flutter (15265): #3      _ReorderableSliverListState._statefulWrap.<anonymous closure>.<anonymous closure> (package:reorderables/src/widgets/reorderable_sliver.dart:844:11)
I/flutter (15265): #4      _DragTargetState.didEnter (package:flutter/src/widgets/drag_target.dart)
I/flutter (15265): #5      _DragAvatar.updateDrag.<anonymous closure> (package:flutter/src/widgets/drag_target.dart:655:23)
I/flutter (15265): #6      _ListBase&Object&ListMixin.firstWhere (dart:collection/list.dart:142:15)
I/flutter (15265): #7      _DragAvatar.updateDrag (package:flutter/src/widgets/drag_target.dart:653:51)
I/flutter (15265): #8      new _DragAvatar (package:flutter/src/widgets/drag_target.dart:589:5)
I/flutter (15265): #9      _DraggableState._startDrag (package:flutter/src/widgets/drag_target.dart:379:35)
I/flutter (15265): #10     LongPressDraggable.createRecognizer.<anonymous closure> (package:flutter/src/widgets/drag_target.dart:317:36)
I/flutter (15265): #11     MultiDragGestureRecognizer._startDrag.<anonymous closure> (package:flutter/src/gestures/multidrag.dart:263:52)
I/flutter (15265): #12     GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:120:24)
I/flutter (15265): #13     MultiDragGestureRecognizer._startDrag (package:flutter/src/gestures/multidrag.dart:263:14)
I/flutter (15265): #14     MultiDragGestureRecognizer.acceptGesture.<anonymous closure> (package:flutter/src/gestures/multidrag.dart:253:48)
I/flutter (15265): #15     _DelayedPointerState.accepted (package:flutter/src/gestures/multidrag.dart:473:14)
I/flutter (15265): #16     MultiDragGestureRecognizer.acceptGesture (package:flutter/src/gestures/multidrag.dart:253:11)
I/flutter (15265): #17     GestureArenaManager._resolveInFavorOf (package:flutter/src/gestures/arena.dart:263:12)
I/flutter (15265): #18     GestureArenaManager._resolve (package:flutter/src/gestures/arena.dart:222:9)
I/flutter (15265): #19     GestureArenaEntry.resolve (package:flutter/src/gestures/arena.dart:52:12)
I/flutter (15265): #20     MultiDragPointerState.resolve (package:flutter/src/gestures/multidrag.dart:62:17)
I/flutter (15265): #21     _DelayedPointerState._delayPassed (package:flutter/src/gestures/multidrag.dart:459:7)
I/flutter (15265): #30     _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
I/flutter (15265): #31     _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
I/flutter (15265): #32     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
I/flutter (15265): (elided 10 frames from class _AssertionError and package dart:async)
I/flutter (15265): 
I/flutter (15265): Handler: onStart
I/flutter (15265): Recognizer:
I/flutter (15265):   DelayedMultiDragGestureRecognizer#0e9ab
I/flutter (15265): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter (15265): Another exception was thrown: NoSuchMethodError: The getter 'width' was called on null.
I/flutter (15265): Another exception was thrown: NoSuchMethodError: The getter 'width' was called on null.
I/flutter (15265): Another exception was thrown: 'package:flutter/src/widgets/sliver.dart': Failed assertion: line 1168 pos 12: '_currentlyUpdatingChildIndex != null': is not true.
I/flutter (15265): Another exception was thrown: 'package:flutter/src/widgets/sliver.dart': Failed assertion: line 1149 pos 12: '_currentlyUpdatingChildIndex == slot': is not true.
I/flutter (15265): Another exception was thrown: 'package:flutter/src/widgets/sliver.dart': Failed assertion: line 1149 pos 12: '_currentlyUpdatingChildIndex == slot': is not true.
I/flutter (15265): Another exception was thrown: 'package:flutter/src/widgets/framework.dart': Failed assertion: line 2271 pos 16: '!_dirtyElements[index]._active || _dirtyElements[index]._debugIsInScope(context)': is not true.
I/flutter (15265): Another exception was thrown: NoSuchMethodError: The getter 'width' was called on null.
I/flutter (15265): Another exception was thrown: 'package:flutter/src/widgets/framework.dart': Failed assertion: line 2989 pos 12: 'child._parent == this': is not true.
I/flutter (15265): Another exception was thrown: 'package:flutter/src/widgets/sliver.dart': Failed assertion: line 1149 pos 12: '_currentlyUpdatingChildIndex == slot': is not true.
I/flutter (15265): Another exception was thrown: 'package:flutter/src/widgets/sliver.dart': Failed assertion: line 1149 pos 12: '_currentlyUpdatingChildIndex == slot': is not true.
`

**Comment by davidBarbieri** [comment]
P.S. i'm using ScopedModel in my project:
```

  @override
  Widget build(BuildContext context) {
    return ScopedModel<RootModel>(
      model: _model,
      child: MaterialApp(
....
....
```

**Comment by hanshengchiu** [comment]
@davidBarbieri I can replicate your error without using ScopedModel.

**Comment by davidBarbieri** [comment]
Thanks man, time to buy you a beer

**Comment by hanshengchiu** [comment]
resolved in v0.2.6.
@davidBarbieri We also offer professional services (Deep Learning, AI, Flutter..). You can check our [website](https://yurusa.com)
